### PR TITLE
Fix empty reply message

### PR DESCRIPTION
**Pull request by jrainville**
Fixes [#7754](https://github.com/status-im/status-desktop/issues/7754)

Uses the already available QuotedMessage data that comes with the Message to show the replied message instead of fetching it using JSON. It's cleaner, less error prone (fixes the issue above) and is way faster.

~~I put this as a Draft because I still need to figure out what to do with deletions of original messages. I can set it as `deleted` when we receive the signal, but if I restart, the message is back, because status-go doesn't handle the og message deletion in the QuotedMessage section.
I need to figure out if that's a bug or if it's intended.~~

The bug has been fixed in status-go. Here is the PR: https://github.com/status-im/status-go/pull/3054

### CODE REVIEW

**Comment by jrainville** [comment]
This is now ready for review

**Review by richard-ramos — APPROVED** [review]

**Review by alaibe — APPROVED** [review]

**Comment by jrainville** [comment]
status-go PR merged. Vendor updated.
@elina2015 you can test

**Comment by glitchminer** [comment]
@jrainville @elina2015 , I had a look at this and found the issue to still occur

image

Steps:
- Have an existing community with messages
- Reply to messages
- Send invite to second user
- Second user accepts
- Second user sees empty reply

**Comment by jrainville** [comment]
I also found a very big issue. If I reply to a very old message that the receiver doesn't know, I just don't receive the reply (nothing at all)

**Comment by jrainville** [comment]
I fixed the issue I mentioned above. It requires this simple status-go PR to be merged: https://github.com/status-im/status-go/pull/3074

**Comment by jrainville** [comment]
@glitchminer I actually had time to work on this, so no need for another PR.

Here is the visual result:
image

Message 1: Normal
Message 2: unknown message, because `old reply` was replying to a **very** old message. The user would need to fetch at the top to hope seeing it
Message 3: the replied message was deleted

**Comment by jrainville** [comment]
@alaibe @richard-ramos I re-requested review because I added a new commit that adds the new `deleted` prop to QuotedMessage. If you want, you can just re-review that new commit. Thanks

**Review by glitchminer — APPROVED** [review]
That's great, thanks for the clarification and the update! I was able to reproduce these and does seem slightly faster as well.

Tested, approved. (Feel free to re-request review for test if second round of code review changes anything).

**Review by alaibe — APPROVED** [review]

File: ui/imports/shared/views/chat/MessageView.qml
Comment (alexjba) [inline ui/imports/shared/views/chat/MessageView.qml:68]: Missing property quotedMessageFromIterator

**Review by alexjba — APPROVED** [review]
Very nice! The only thing I think we also need to include in the message_model is the contact details of the quoted message sender. Will do it once I merge #9031 with this PR

**Comment by anastasiyaig** [comment]
@jrainville seems to be fine to merge (once conflicts are resolved) 🙏

**Pull request merged**
